src/rules/extensions.js

```diff
@@ -225,13 +231,32 @@ module.exports = {
             node: source,
             message:
               `Missing file extension ${extension ? `"${extension}" ` : ''}for "${importPathWithQueryString}"`,
+            ...props.fix && extension ? {
+              fix(fixer) {
+                return fixer.replaceText(
+                  source,
+                  JSON.stringify(`${importPathWithQueryString}.${extension}`),
```

Comment:
See https://github.com/un-ts/eslint-plugin-import-x/pull/329/files#r2094349587

> `JSON.stringify` is not correct for single quoted imports.

https://github.com/un-ts/eslint-plugin-import-x/pull/327/files#r2094298890

`importPathWithQueryString` could contains `query` and `hash`, and could also ends with `.`, `..` or `/`.